magma/passes/drive_undriven.py

```diff
@@ -2,12 +2,20 @@
 from ..is_definition import isdefinition
 
 
+def _drive_if_undriven_input(port):
+    if port.is_mixed():
+        # Sum so it doesn't short circuit
+        return sum(_drive_if_undriven_input(p) for p in port)
```

Comment:
```suggestion
        undrivens = (_drive_if_undriven_input(p) for p in port)
        return any(undrivens)
```

This is a little more readable

Reply:
Any doesn't work because it short circuits

Reply:
Used list comp to prevent short circuiting.